ast/compile.go

```diff
@@ -3036,7 +3036,16 @@ func outputVarsForExpr(expr *Expr, arity func(Ref) int, safe VarSet) VarSet {
 
 		return outputVarsForExprCall(expr, ar, safe, terms)
 	case *Every:
-		return outputVarsForTerms(terms.Domain, safe)
+		s := outputVarsForTerms(terms.Domain, safe)
+
+		cpy := safe.Copy()
+		if terms.Key != nil {
+			cpy.Add(terms.Key.Value.(Var))
+		}
+		cpy.Add(terms.Value.Value.(Var))
+
+		s.Update(outputVarsForBody(terms.Body, arity, cpy))
```

Comment:
I'm surprised that we're adding the output vars from the body of the Every to the set returned here. Won't that make vars in the body of the every appear safe in the outside body? E.g.,

```
every x in y { z = x }
z > 10  # won't z be safe here? shouldn't it be unsafe?
```

Reply:
Yeah that's a good point. I don't know what I was thinking, but I'll dial that back. Thanks!